tests/e2e/initialization/helper.go

```diff
@@ -0,0 +1,137 @@
+// Where `cdc` is client codec, `genesisFileUrl` is the path/url of current genesis file,
+// `accAddr` is the address to be added to the genesis state, `amountStr` is the list of initial coins
+// to be added for the account, `appendAcct` updates the account if already exists.
+// `vestingStart, vestingEnd and vestingAmtStr` respectively are the schedule start time, end time (unix epoch)
+// and coins to be appended to the account already in the genesis.json file.
+func AddGenesisAccount(
+	cdc codec.Codec,
+	accAddr sdk.AccAddress,
+	appendAcct bool,
+	genesisFileUrl, amountStr, vestingAmtStr string,
+	vestingStart, vestingEnd int64,
+) error {
+	coins, err := sdk.ParseCoinsNormalized(amountStr)
+	if err != nil {
+		return fmt.Errorf("failed to parse coins: %w", err)
+	}
+
+	vestingAmt, err := sdk.ParseCoinsNormalized(vestingAmtStr)
+	if err != nil {
+		return fmt.Errorf("failed to parse vesting amount: %w", err)
+	}
+
+	// create concrete account type based on input parameters
+	var genAccount authtypes.GenesisAccount
+
+	balances := banktypes.Balance{Address: accAddr.String(), Coins: coins.Sort()}
+	baseAccount := authtypes.NewBaseAccount(accAddr, nil, 0, 0)
+
+	if !vestingAmt.IsZero() {
+		baseVestingAccount := authvesting.NewBaseVestingAccount(baseAccount, vestingAmt.Sort(), vestingEnd)
+
+		if (balances.Coins.IsZero() && !baseVestingAccount.OriginalVesting.IsZero()) ||
+			baseVestingAccount.OriginalVesting.IsAnyGT(balances.Coins) {
+			return errors.New("vesting amount cannot be greater than total amount")
+		}
+
+		switch {
+		case vestingStart != 0 && vestingEnd != 0:
+			genAccount = authvesting.NewContinuousVestingAccountRaw(baseVestingAccount, vestingStart)
+
+		case vestingEnd != 0:
+			genAccount = authvesting.NewDelayedVestingAccountRaw(baseVestingAccount)
+
+		default:
+			return errors.New("invalid vesting parameters; must supply start and end time or end time")
+		}
+	} else {
+		genAccount = baseAccount
+	}
+
+	if err := genAccount.Validate(); err != nil {
+		return fmt.Errorf("failed to validate new genesis account: %w", err)
+	}
+
+	appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genesisFileUrl)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+	}
+
+	authGenState := authtypes.GetGenesisStateFromAppState(cdc, appState)
+
+	accs, err := authtypes.UnpackAccounts(authGenState.Accounts)
+	if err != nil {
+		return fmt.Errorf("failed to get accounts from any: %w", err)
+	}
+
+	bankGenState := banktypes.GetGenesisStateFromAppState(cdc, appState)
+	if accs.Contains(accAddr) {
+		if !appendAcct {
+			return fmt.Errorf("Account %s already exists\nUse `append` flag to append account at existing address", accAddr)
+		}
+
+		genesisB := banktypes.GetGenesisStateFromAppState(cdc, appState)
+		for idx, acc := range genesisB.Balances {
+			if acc.Address != accAddr.String() {
+				continue
+			}
+
+			updatedCoins := acc.Coins.Add(coins...)
```

Comment:
Wdyt: doesn't it make more sense to use `amountStr` as override parameter here and not to add it to existing coins? I think it makes more sense because when adding coins to existing account, you won't need to think about current balance and just specify the desired in `amountStr` parameter. This is just an idea, we will see if others like it

Reply:
Thank you for your suggestion, I think we should still keep this with precise logic, btw this is the draft version of this pr, in the future I will call this function from osmosis/cosmos-sdk in this link https://github.com/osmosis-labs/cosmos-sdk/blob/osmosis-main/x/auth/helpers/genaccounts.go

Reply:
And can you explain more about your suggestion? I don't really understand all of it

Reply:
I do not know if that is possible what I am suggesting, but what I mean is: do not add `coins` to account balance if that account exists, but override the current balance of this account with `coins` provided

lets say we have an account with `100uosmo`. If we are adding this account `AddGenesisAccount`, and provide `amountStr` as `120uosmo`, we end up with `220uosmo` because they are just added. My suggestion is this: if you provide `120uosmo` as `amountStr`, it essentially overrides the balance, so after this function, account that initially had `100uosmo` will have `120uosmo`. I think this way if a user wants a specific account to have some specific amount, they do not need to think about the current balance. Lmk if that makes sense

Reply:
I think it does not override the balances instead add-in

Reply:
yeah that is true I am just suggesting the override approach